ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivDebugGetStateRoot.java

```diff
@@ -74,6 +75,20 @@ protected Object resultByBlockNumber(
     } catch (final MultiTenancyValidationException e) {
       return new JsonRpcErrorResponse(
           requestContext.getRequest().getId(), FIND_PRIVACY_GROUP_ERROR);
+    } catch (final EnclaveClientException e) {
+      if (e.getMessage().equals(JsonRpcError.ENCLAVE_PRIVACY_GROUP_MISSING.getMessage())
+          || (e.getMessage().contains("Privacy group")
+              && e.getMessage()
+                  .contains("not found"))) { // TODO: In Tessera we are throwing this: new
+        // PrivacyGroupNotFoundException("Privacy group " +
```

Comment:
This is a bit brittle. But if all we have from tessera is this error string than it is better than nothing. Parsing using a regexp `Privacy group [^ ] not found` would be a bit tighter.

Reply:
I've just checked what the message is that I get, and it looks like this: "Privacy group] not found"
Not sure where that comes from ...
I guess I should create a ticket to fix that in Tessera. The PrivacyGroup.Id is an Object and not a String, but it does have a getBase64() method. It does have a toString(), but that does funny things.